integration: Modify branch names for Bitbucket 3 events.

Joelute opened the pull request:
This is a follow up to https://github.com/zulip/zulip/pull/24673, we want to modify every webhook events to follow the same pattern and consistency where branch name should only show on opened and merged events.

In this PR there are two prep commits where:
- We move assignee message to the end for git integrations.
- Allow `get_pull_request_event_message` to support reviewers.

In the main changes we have:
- Removed branch names for updated PR events.
- Added branch names for merged PR events.


<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
Will add screenshots and screen captures later.
<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented:
It looks like you created a block without test coverage? Is it easy to fix that?

Joelute commented:
I'm actually having troubles fixing it. I don't think it's a hard fix but I don't know where the test files for library functions are located.

Joelute commented:
Updated the commits to fix the failing test cases. Previously I created an additional block of that was used yet, my thought process was that it would be useful in the future commits. For now I've removed those blocks as they aren't being used and it's not being tested.

timabbott commented:
Nice, merged, thanks @Joelute!
